### DISCUSSION TITLE
Battery life

### DISCUSSION THREAD

**Discussion by O491dogan** [wottreng/Linux-Mint-on-Lenovo-Legion-5#3] [Q&A] [answered]
Hey! 
Thanks for your write up and your scripts! :-)

As I've bought a Legion 5 too, but I'm currently not very satisfied with battery life on Linux Mint. 
I don't use too much of display brightness where possible and I'm running on the internal graphics (at least this is shown on the applet in the taskbar). 
Battery life is about 2.5 hours doing nothing whereas under Windows I get 7+ hours (same scenario, idling in desktop).
I know tools like TLP and Slimbook Battery exist - I just wanted to ask if you have already estimated some good settings.

Thank you very much!

**Comment by wottreng** [comment]
I currently get the same. I just installed powertop (sudo apt install powertop) and its reporting 24 W of power usage. That is definitely too high. should be closer to 15 W, I believe. 
powertop offers many tunables so Ill have to play with it to see what works.
If you find something let me know and Ill add it to the repository.
Thanks for your comment!

**Comment by O491dogan** [comment, marked as answer]
Hi again, thank you for your response!

I've got some news about this topic.
I had a closer look on the the devices with powertop. It seems like the dGPU is drawing power even when not in use (I usually use the 'NVIDIA on demand' preset).
As soon as I tuned the value with powertop, the power draw dropped to around 10 Watts. I then noted down the command and added it as a startup command when logging in.
`echo 'auto' > /sys/bus/pci/devices/0000:01:00.0/power/control`
Edit: The default value is 'on'.

This boosts battery life quite well!

I haven't yet tested how it turns out when the dGPU should be used. Because I have a dual boot setup I've already noticed under Windows 10 that AFTER the dGPU has been in use and the nvidia driver/mux switches back to the iGPU, the dGPU still draws power, but is not in use anymore.
I'll report here if I have some more findings on that.

**Comment by O491dogan** [comment]
Okay, I used Handbrake with NVENC to convert a media file. 
Driver used: 510.47.03
Kernel used: 5.13.0.35

Handbrake utilized the video core of the dGPU to convert the file. After that the power consumption went back to ~ 10 Watts and fans went off after about 2 minutes of cooling the system down again. This seems to work - but I haven't used the gpu core yet.

**Comment by wottreng** [comment]
Awesome! I got my power down to 10w as well just using `nvidia-settings` and setting dGPU to on-demand. I confirmed that it was off with `cat /sys/bus/pci/devices/0000:01:00.0/power/control` -> `auto`. Is your nvidia PRIME profile set to on-demand as well as your bios set to adaptive dGPU? I updated the issue associated with this to reflect how to fix.  Ill update the readME soon. Thanks for your input!

**Reply by O491dogan** [reply]
Hey!
Yes - my BIOS settings are adaptive or dynamic dGPU and my NVIDIA PRIME profile is set to on-demand. 

What I wanted to add is the following:
I'm currently using **Kernel 5.16.12-051612-generic**. I'm using it because the brightness-keys work out of the box. 
What I recognize is that the nvidia driver is not loaded because there are some missing headers and fw packages to build the initram image. I'm by faaaaar not an expert and do not understand the Linux 'under the hood' but I'm very confident, that there is going to be a way that everything works nicely together, when the ne LM version comes out in April. 
I think it will use Kernel 5.13.something - but we may be able to upgrade to sth. newer easily. 

What I noticed as well is that the stepsize of LM regarding the brightness is 5 percent by default. In the evening, 5 percent is to bright for my eyes so I punched a bash script together, where I can adjust the brightness in 1 percent steps if the current brightness is 5 percent or lower. This does not work with the brightness keys but with some "starters" in the taskbar. If you have interest, just let me know.

The thing with Kernel 5.16.x right now is that I'm not able to use the dGPU because of the driver not loading. But for me this is not a big deal right now because I only use the iGPU with LM and the dGPU on ... well the other well known OS ;-).

Sorry for mixing up many topics in this 'battery life' section!

Thank you very much for putting in the time to create and update the guides! :-)
